samples/bot-sso-adaptivecard/nodejs/bots/botSSOAdaptiveCard.js

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { TeamsActivityHandler, CardFactory, ActivityHandler } = require('botbuilder');
+const AdaptiveCardResponse = require('../resources/adaptiveCardResponseJson.json');
+const AdaptiveCardWithSSO = require('../resources/AdaptiveCardWithSSOInRefresh.json');
+const Options = require('../resources/options.json');
+var ACData = require("adaptivecards-templating");
+
+class BotSSOAdativeCard extends TeamsActivityHandler {
+    constructor() {
+        super();
+
+        this.onMembersAdded(async (context, next) => {
+            //Send welcome message when app installed
+            await this.sendWelcomeMessage(context);
+
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+
+        this.onMessage(async (context, next) => {
+            const text = context.activity.text;
+
+            if (text) {
+                switch (text) {
+                    case "login":
+                        await context.sendActivity({ attachments: [this.getInitialAdaptiveCard(),] });
+                        break;
+
+                    case "PerformSSO":
+                        await context.sendActivity({ attachments: [this.getAdaptiveCardWithSSO()] });
+                        break;
+
+                    default:
+                        const message = "Please send 'login' for options";
+                        await context.sendActivity(message);
+                        break;
+                }
+            }
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+    }
+
+    /**
+    * Invokes when Action.Submit button performs onclick action. 
+    * @param {turnContext} context
+    */
+    async onInvokeActivity(context) {
+        if (context._activity.name == "adaptiveCard/action") {
+            if (context._activity.value == null) {
+                return null;
+            }
+
+            var value = context._activity.value;
+
+            if (value["action"] == null) {
+                return null;
+            }
+
+            var actiondata = value["action"];
+
+            if (actiondata["verb"] == null) {
+                return null;
+            }
+
+            // authToken and state are absent, handle verb
+            let verb = actiondata["verb"];
+
+            var authentication = null;
+
+            // When adaptiveCard / action invoke activity from teams contains token in response to sso flow from earlier invoke.
+            if (value["authentication"] != null) {
+                authentication = value["authentication"];
+            }
+
+            if (authentication == null) {
+                // authToken and state are absent, handle verb
```

Comment:
where are you checking for state?

Reply:
As per your below comment inside authentication, Token is present bot not state..
<img width="718" alt="image" src="https://user-images.githubusercontent.com/109147706/218767903-5df82fee-7598-43f0-a8e1-c3e005374482.png">